src/nvim/path.c

```diff
@@ -359,8 +359,12 @@ char *concat_fnames(const char *fname1, const char *fname2, bool sep)
 void add_pathsep(char *p)
   FUNC_ATTR_NONNULL_ALL
 {
-  if (*p != NUL && !after_pathsep(p, p + strlen(p)))
-    strcat(p, PATHSEPSTR);
+  const size_t len = strlen(p);
+  if (*p != NUL && !after_pathsep(p, p + len) && len < MAXPATHL - 1) {
+    memcpy(p + len, PATHSEPSTR, sizeof(PATHSEPSTR));
+  } else if (len >= MAXPATHL - 1) {
+    EMSG(_(e_pathtoolong));
```

Comment:
Callers of `add_pathsep()` may also print `EMSG()`. Small utility functions like this should return a `bool`, or arguably even `strlen(p) + 1`.

Notably, this change makes it an implicit precondition that `p` is a `MAXPATHL`-sized buffer.